src/panel-app/useImportMapOverrides.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { evalCmd } from "../inspected-window.helper.js";
+
+export default function useImportMapOverrides() {
+  const [importMapsEnabled, setImportMapEnabled] = useState(false);
+  const [overrides, setOverrides] = useState({});
+
+  async function checkImportMapOverrides() {
+    const hasImportMapsEnabled = await evalCmd(`(function() {
+      return !!window.importMapOverrides
+    })()`);
+    setImportMapEnabled(hasImportMapsEnabled);
+  }
+
+  async function getImportMapOverrides() {
+    const { imports } = await evalCmd(`(function() {
+      return window.importMapOverrides.getOverrideMap()
+    })()`);
+    setOverrides(imports);
+  }
+
+  async function addOverride(currentMap, currentUrl) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.addOverride("${currentMap}", "${currentUrl}")
+    })()`);
+  }
+
+  async function removeOverride(currentMap) {
+    await evalCmd(`(function() {
+      return window.importMapOverrides.removeOverride("${currentMap}")
+    })()`);
+  }
+
+  async function batchSetOverrides() {
+    const overrideCalls = Object.entries(overrides).map(([map, url]) =>
+      !url ? removeOverride(map) : addOverride(map, url)
+    );
+    await Promise.all(overrideCalls);
+    await evalCmd(`window.location.reload()`);
+  }
+
+  // Get initial list of maps if they exist
+  useEffect(() => {
+    const hasImportMapsEnabled = checkImportMapOverrides();
+    if (hasImportMapsEnabled) {
+      getImportMapOverrides();
+    }
+  }, []);
+
+  const setOverride = (mapping, url) => {
+    const newMapping = { [mapping]: url };
+    const newOverrides = Object.assign({}, overrides, newMapping);
```

Comment:
ooh baby, like it.

could also use object spread here if you wanted: 

```jsx
const newOverrides = {
  ...overrdies,
  [mapping]: url
}
```

but it's all the same. :)

Reply:
Only reason I did this is because its slightly clearer, and this style of explicitness helps me when coming back to a codebase. I don't expect that this will change much but might help other contributors.

Reply:
Ehh... looking at this again, its not all that much clearer and the spread syntax is much more popular than Object.assign. I'm going to change this one.